src/runtime/Types/MetaType.cs

```diff
@@ -79,49 +82,56 @@ public static NewReference tp_new(BorrowedReference tp, BorrowedReference args,
             BorrowedReference bases = Runtime.PyTuple_GetItem(args, 1);
             BorrowedReference dict = Runtime.PyTuple_GetItem(args, 2);
 
-            // We do not support multiple inheritance, so the bases argument
-            // should be a 1-item tuple containing the type we are subtyping.
-            // That type must itself have a managed implementation. We check
-            // that by making sure its metatype is the CLR metatype.
+            // Extract interface types and base class types.
+            var interfaces = new List<Type>();
+            var baseType = new List<ClassBase>();
 
-            if (Runtime.PyTuple_Size(bases) != 1)
+            var cnt = Runtime.PyTuple_GetSize(bases);
+
+            for (uint i = 0; i < cnt; i++)
             {
-                return Exceptions.RaiseTypeError("cannot use multiple inheritance with managed classes");
+                var base_type2 = Runtime.PyTuple_GetItem(bases, (int)i);
+                var cb2 = (ClassBase) GetManagedObject(base_type2);
+                if (cb2 != null)
+                {
+                    if (cb2.type.Valid && cb2.type.Value.IsInterface)
+                        interfaces.Add(cb2.type.Value);
+                    else baseType.Add(cb2);
```

Comment:
IMHO, it is better to enforce the order to be `BaseClass, IInterface1, IInterface2`, etc.

I believe the current code has a bug down below because it uses `base_type` which is not the `BaseClass` if the order is different.

Reply:
I have only added this functionality for types which goes through  TypeManager.CreateSubType, so it should only affect that branch.

for code flows not invoking CreateSubType, the behavior willl be exactly the same as before.